tools/StaticAnalysis/Exceptions/Az.Network/SignatureIssues.csv

```diff
@@ -293,3 +293,6 @@
 "Microsoft.Azure.PowerShell.Cmdlets.Network.dll","Microsoft.Azure.Commands.Network.UpdateAzureRmExpressRouteGatewayCommand","Set-AzExpressRouteGateway","1","8410","Parameter MaxScaleUnits of cmdlet Set-AzExpressRouteGateway does not follow the enforced naming convention of using a singular noun for a parameter name.","Consider using a singular noun for the parameter name."
 "Microsoft.Azure.PowerShell.Cmdlets.Network.dll","Microsoft.Azure.Commands.Network.NewNetworkWatcherNetworkConfigurationDiagnosticProfileCommand","New-AzNetworkWatcherNetworkConfigurationDiagnosticProfile","1","8100","New-AzNetworkWatcherNetworkConfigurationDiagnosticProfile Does not support ShouldProcess but the cmdlet verb New indicates that it should.","Determine if the cmdlet should implement ShouldProcess and if so determine if it should implement Force / ShouldContinue"
 "Microsoft.Azure.PowerShell.Cmdlets.Network.dll","Microsoft.Azure.Commands.Network.InvokeNetworkWatcherNetworkConfigurationDiagnosticCommand","Invoke-AzNetworkWatcherNetworkConfigurationDiagnostic","1","8100","Invoke-AzNetworkWatcherNetworkConfigurationDiagnostic Does not support ShouldProcess but the cmdlet verb Invoke indicates that it should.","Determine if the cmdlet should implement ShouldProcess and if so determine if it should implement Force / ShouldContinue"
+"Microsoft.Azure.PowerShell.Cmdlets.Network.dll","Microsoft.Azure.Commands.Network.GetAzureApplicationGatewayAvailableRequestHeaders","Get-AzApplicationGatewayAvailableRequestHeaders","1","8400","Get-AzApplicationGatewayAvailableRequestHeaders uses the noun 'AzApplicationGatewayAvailableRequestHeaders', which does not follow the enforced naming convention of using a singular noun for a cmdlet name.","Consider using a singular noun for the cmdlet name."
```

Comment:
@nirkum18 these exceptions shouldn't be suppressed, you should have the noun name for the cmdlets be singular to follow the singularity rule in PowerShell: https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/azure-powershell-design-guidelines.md#noun-singularity

Reply:
@cormacpayne new cmdlets are similar to `Get-AzApplicationGatewayAvailableSslOptions` and `Get-AzApplicationGatewayAvailableWafRuleSets`. They always return array of predefined values, so I think it falls under

> Plural parameter names should be used only in those cases where the value of the parameter is always a multiple-element value.

Reply:
@cormacpayne as @number213 mentioned above, these commandlets return an array of strings, so it should fall under plural naming.

Reply:
@nirkum18 @number213 I either misspoke or PowerShell recently updated their patterns 😁 It looks like they advise following the singular noun rule for cmdlets in _all_ cases:

From the [_Strongly Encouraged Development Guidelines_](https://docs.microsoft.com/en-us/powershell/developer/cmdlet/strongly-encouraged-development-guidelines#use-a-specific-noun-for-a-cmdlet-name-sd01):

_To enhance the user experience, the noun that you choose for a cmdlet name should be singular. For example, use the name Get-Process instead of Get-Processes. It is best to follow this rule for all cmdlet names, even when a cmdlet is likely to act upon more than one item._

Reply:
@cormacpayne I have updated the PR with singular naming convention.